lib/model/message.dart

```diff
@@ -168,10 +181,26 @@ class MessageStoreImpl with MessageStore {
       return;
     }
 
-    // final newStreamId = event.newStreamId; // null if topic-only move
+    final newChannelId = event.newStreamId; // null if topic-only move
+
+    // Eliminate the case when the topic is updated only to be resolved/unresolved.
+    // return early as the messages' editState need no update.
```

Comment:
I find it's often helpful to put this sort of comment inside the `if` body, rather than above the `if`. That helps it get straight to the point with less need to repeat what the `if` condition says, because the comment's location means it's talking about the case where that condition is true.

So here:
```dart
    if (newChannelId == null
        && MessageEditState.areSameTopic(event.origTopic!, event.newTopic!)) {
      // The topic was only resolved or unresolved.
      // No change to the messages' editState.
      return;
    }
```

I guess that also illustrates another way it helps make the comment more focused: it has less need to repeat what the body of the `if` says either, because the body is right there. Between the two, the comment gets to avoid repeating the "what" (that the code already expresses) and can focus only on the "why".